aten/src/ATen/native/cuda/Shape.cu

```diff
@@ -159,9 +165,22 @@ void parallel_cat(Tensor &out, const TensorList &inputs, int64_t dimension,
   OutputTensorSizeStride<unsigned int, CAT_ARRAY_MAX_INPUT_DIMS> param;
 
   // Next, let's initialize the size, stride arrays for the output Tensor.
-  for (int i = 0; i < nDims; ++i) {
-    param.outputSize[i] = at::native::size(out, i);
-    param.outputStride[i] = out.stride(i);
+  if (memory_format == c10::MemoryFormat::Contiguous) {
+    for (int i = 0; i < nDims; ++i) {
+      param.outputSize[i] = at::native::size(out, i);
+      param.outputStride[i] = out.stride(i);
+    }
+  } else {
```

Comment:
Can you have an `elseif` for channels-last memory format here, and TORCH_CHECK with "unsupported memory format" in the `else` branch.

Reply:
fixed